zerver/lib/actions.py

```diff
@@ -4866,10 +4870,12 @@ def do_update_user_custom_profile_data(user_profile: UserProfile,
     with transaction.atomic():
         update_or_create = CustomProfileFieldValue.objects.update_or_create
         for field in data:
-            update_or_create(user_profile=user_profile,
-                             field_id=field['id'],
-                             defaults={'value': field['value']})
-            notify_user_update_custom_profile_data(user_profile, field)
+            field_value, created = update_or_create(user_profile=user_profile,
+                                                    field_id=field['id'],
+                                                    defaults={'value': field['value']})
+            notify_user_update_custom_profile_data(user_profile, {"id": field['id'],
+                                                                  "value": field['value'],
+                                                                  "type": field_value.field.field_type})
```

Comment:
I think it might be cleaner to have the arguments to `notify_user_update_custom_profile_data` be based on e.g. `field_value.value`, so we're pulling from a consistent data source rather than a mix of the arguments and the return value from `update_or_create`?

Reply:
I updated it to dump the value if its list before passing to `notify_user_update_custom_profile_data` function. 
Its not possible to find type of field with `field_value.value`, we require `field` object correspond to it, which can get from `update_or_create` return argument. Or we can just convert list type values to json object like above.